studygroups/views/api.py

```diff
@@ -229,6 +231,15 @@ def get(self, request):
                 default=F('first_meeting_date_value'),
                 output_field=CharField(),
             )
+        ).annotate(
+            status=Case(
+                When(signup_open=True, first_meeting_date__gt=today, then=Value('upcoming')),
+                When(signup_open=True, first_meeting_date__lte=today, last_meeting_date__gte=today, then=Value('in_progress')),
+                When(Q(signup_open=False) | Q(last_meeting_date__lt=today), then=Value('completed')),
```

Comment:
Why does this or on `signup_open == False`? Shouldn't any past learning circle be completed even if signup is open. And closed on if signup is closed for an 'in progress' learning circle?

Reply:
oh yes you're right, i fixed that and added a test.